packages/compass-schema/src/stores/reducer.ts

```diff
@@ -0,0 +1,312 @@
+    const { schema } = getState();
+    const hasSchema = schema !== null;
+    if (hasSchema) {
+      void navigator.clipboard.writeText(JSON.stringify(schema, null, '  '));
+    }
+    dispatch(_trackSchemaShared(hasSchema));
+    openToast(
+      'share-schema',
+      hasSchema
+        ? {
+            variant: 'success',
+            title: 'Schema Copied',
+            description: `The schema definition of ${namespace} has been copied to your clipboard in JSON format.`,
+            timeout: 5_000,
+          }
+        : {
+            variant: 'warning',
+            title: 'Analyze Schema First',
+            description: 'Please Analyze the Schema First from the Schema Tab.',
+            timeout: 5_000,
+          }
+    );
+  };
+};
+
+export const _trackSchemaShared = (
+  hasSchema: boolean
+): SchemaThunkAction<void> => {
+  return (dispatch, getState, { track, connectionInfoRef }) => {
+    const { schema } = getState();
+    // Use a function here to a) ensure that the calculations here
+    // are only made when telemetry is enabled and b) that errors from
+    // those calculations are caught and logged rather than displayed to
+    // users as errors from the core schema sharing logic.
+    const trackEvent = () => ({
+      has_schema: hasSchema,
+      schema_width: schema?.fields?.length ?? 0,
+      schema_depth: schema ? calculateSchemaDepth(schema) : 0,
+      geo_data: schema ? schemaContainsGeoData(schema) : false,
+    });
+    track('Schema Exported', trackEvent, connectionInfoRef.current);
+  };
+};
+
+/**
+ * @return {Object} initial schema state.
+ */
+const getInitialState = (): SchemaState => ({
+  analysisState: ANALYSIS_STATE_INITIAL,
+  errorMessage: '',
+  schema: null,
+  resultId: resultId(),
+});
+
+export const geoLayerAdded = (
+  field: string,
+  layer: Layer
+): SchemaThunkAction<ReturnType<typeof generateGeoQuery>> => {
+  return (dispatch, getState, { geoLayersRef }) => {
+    geoLayersRef.current = addLayer(
+      field,
+      layer as Circle | Polygon,
+      geoLayersRef.current
+    );
+    return generateGeoQuery(geoLayersRef.current);
+  };
+};
+
+export const geoLayersEdited = (
+  field: string,
+  layers: LayerGroup
+): SchemaThunkAction<ReturnType<typeof generateGeoQuery>> => {
+  return (dispatch, getState, { geoLayersRef }) => {
+    layers.eachLayer((layer) => {
+      dispatch(geoLayerAdded(field, layer));
+    });
+    return generateGeoQuery(geoLayersRef.current);
+  };
+};
+
+export const geoLayersDeleted = (
+  layers: LayerGroup
+): SchemaThunkAction<ReturnType<typeof generateGeoQuery>> => {
+  return (dispatch, getState, { geoLayersRef }) => {
+    layers.eachLayer((layer) => {
+      delete geoLayersRef.current[(layer as any)._leaflet_id];
+    });
+    return generateGeoQuery(geoLayersRef.current);
+  };
+};
+
+export const stopAnalysis = (): SchemaThunkAction<void> => {
+  return (dispatch, getState, { abortControllerRef }) => {
+    if (!abortControllerRef) return;
```

Comment:
Based on types and the plugin setup code this value is always there, so this check is not doing anything, what's the purpose?